src/HealthGPS/analysis_module.cpp

```diff
@@ -560,6 +632,44 @@ void AnalysisModule::initialise_output_channels(RuntimeContext &context) {
     channels_.emplace_back("std_yld");
     channels_.emplace_back("mean_daly");
     channels_.emplace_back("std_daly");
+
+    // Since we will be performing frequent lookups, we will store the strings and indexes in a map
+    // for quick access.
+    for (size_t i = 0; i < channels_.size(); i++) {
+        channel_index_.emplace(channels_[i], i);
+    }
+}
+
+size_t AnalysisModule::calculate_index(const Person &person) const {
```

Comment:
Could you maybe add a unit test for this function to `HealthGPS.Tests`? I've been pretty bad at writing tests for HealthGPS myself, but this one should be easy enough to write a test for and then we could be a bit more confident it's doing the right thing.

Reply:
I have created an issue addressing this: https://github.com/imperialCHEPI/healthgps/issues/470